pytest_fixtures/component/virtwho_config.py

```diff
@@ -350,3 +351,15 @@ def delete_host(form_data_api, target_sat):
     results = target_sat.api.Host().search(query={'search': guest_name})
     if results:
         target_sat.api.Host(id=results[0].read_json()['id']).delete()
+
+
+@pytest.fixture
+def register_sat_and_enable_aps_repo(target_sat):
+    """Register Satellite to CDN and Enable rhel aps repos"""
+    target_sat.register_to_cdn()
+    if target_sat.os_version.major == 8:
+        target_sat.enable_repo(REPOS['rhel8_aps']['id'])
+    elif target_sat.os_version.major == 9:
+        target_sat.enable_repo(REPOS['rhel9_aps']['id'])
+    else:
+        pass
```

Comment:
Yet one small proposal that might work RHEL.next too
```suggestion
    target_sat.enable_repo(REPOS[f'rhel{target_sat.os_version.major}_aps']['id'])
```